src/test/java/de/gessnerfl/fakesmtp/controller/EmailRestControllerMVCIntegrationTest.java

```diff
@@ -179,6 +181,100 @@ void shouldDeleteAllEmails() throws Exception {
         assertThat(emailRepository.findAll(), empty());
     }
 
+    @Test
+    void shouldSearchEmailsByToAddress() throws Exception {
+        createRandomEmails(5, 1);
+
+        MvcResult mvcResult = this.mockMvc.perform(get("/api/emails/search?query=toAddress:receiver@example.com"))
+                .andReturn();
+
+        @SuppressWarnings("unchecked")
+        ArrayList<Email> emailSearch = mapFromJson(mvcResult.getResponse().getContentAsString(), ArrayList.class);
+
+        assertEquals(emailSearch.size(), 5);
+    }
+
+    @Test
+    void shouldSearchEmailsById() throws Exception {
+        createRandomEmails(5, 1);
+        var email = createRandomEmail(5);
+
+        MvcResult mvcResult = this.mockMvc.perform(get("/api/emails/search?query=id:" + email.getId()))
+                .andReturn();
+
+        ArrayList<Email> emailSearch = mapFromJson(mvcResult.getResponse().getContentAsString(), new TypeReference<ArrayList<Email>>() {});
+
+        assertEquals(emailSearch.size(), 1);
+        assertEquals(email.getId(), emailSearch.get(0).getId());
+        assertEquals(email.getFromAddress(), emailSearch.get(0).getFromAddress());
+        assertEquals(email.getMessageId(), emailSearch.get(0).getMessageId());
+        assertEquals(email.getSubject(), emailSearch.get(0).getSubject());
+        assertEquals(email.getToAddress(), emailSearch.get(0).getToAddress());
+    }
+
+    @Test
+    void shouldSearchEmailsByToAddressRegex() throws Exception {
+        createRandomEmails(5, 1);
+
+        MvcResult mvcResult = this.mockMvc.perform(get("/api/emails/search?query=toAddress:*receiver*"))
+                .andReturn();
+
+        ArrayList<Email> emailSearch = mapFromJson(mvcResult.getResponse().getContentAsString(), new TypeReference<ArrayList<Email>>() {});
+
+        assertEquals(emailSearch.size(), 5);
+    }
+
+    @Test
+    void shouldSearchEmailsBySubjectRegexAndToAddress() throws Exception {
+        createRandomEmails(5, 1);
+        var email1 = save(EmailControllerUtil.prepareEmail("subject", "friend@address.domain", 1));
+        var email2 = save(EmailControllerUtil.prepareEmail("an another subject", "friend@address.domain", 15));
+
+        MvcResult mvcResult = this.mockMvc.perform(get("/api/emails/search?query=subject:*subject*,AND,toAddress:friend@address.domain"))
```

Comment:
The comma between the expression/logical operator looks quite strange. Is it to avoid URL encoding for space?

Meaning `/api/emails/search?query=%2Asubject%2A%20AND%20toAddress%3Afriend%40address.domain`